test/src/test/java/hudson/security/HudsonPrivateSecurityRealmTest.java

```diff
@@ -342,6 +346,41 @@ public void userCreationWithHashedPasswords() throws Exception {
         assertTrue(spySecurityListener.createdUsers.get(0).equals("charlie_hashed"));
     }
 
+    @Issue("JENKINS-56008")
+    @Test
+    public void updateUserPassword() throws Exception {
+        HudsonPrivateSecurityRealm securityRealm = new HudsonPrivateSecurityRealm(true, false, null);
```

Comment:
Not sure if you've tried these before, but perhaps you can simplify these tests a bit by using `JenkinsRule.createDummySecurityRealm()`? I haven't tried it myself, though I have seen many tests use it.

Reply:
hmh.. I've tried it before in some previous tests, but wasn't sure how well it mimics a real user account compared to creating one manually. Will take another look at it though.

Reply:
I looked into this, reference here - https://gitter.im/jenkinsci/outreachy?at=5c6c4c58ab952d30855a538e